libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,221 @@ closure color volume_matte() BUILTIN;
+//  \param  transmission_tint   Weight per color channel for the transmission lobe. Set to (0,0,0) to disable transmission.
+//  \param  roughness_x         Surface roughness in the U direction. Valid range [0,1] with a perceptually linear response over the range.
+//  \param  roughness_y         Surface roughness in the V direction. Valid range [0,1] with a perceptually linear response over the range.
+//  \param  f0                  Reflectivity per color channel at facing angles.
+//  \param  f90                 Reflectivity per color channel at grazing angles.
+//  \param  distribution        Microfacet distribution. An implementation is expected to support the following distributions: { "ggx" }
+//  \param  thinfilm_thickness  Optional float parameter for thickness of an iridescent thin film layer on top of this BSDF. Given in nanometers.
+//  \param  thinfilm_ior        Optional float parameter for refraction index of the thin film layer.
+//  \param  label               Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color generalized_schlick_bsdf(normal N, vector U, color reflection_tint, color transmission_tint, float roughness_x, float roughness_y, color f0, color f90, float exponent, string distribution) BUILTIN;
+​
+// Constructs a translucent (diffuse transmission) BSDF based on the Lambert reflectance model.
+//
+//  \param  N           Normal vector of the surface point beeing shaded.
+//  \param  albedo      Surface albedo.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color translucent_bsdf(normal N, color albedo) BUILTIN;
+
+// Constructs a closure that represents straight transmission through a surface.
+//
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+// NOTE:
+//  - This is not a node in the MaterialX library, but the surface shader constructor
+//    node has an 'opacity' parameter to control textured cutout opacity.
+//
+closure color transparent_bsdf() BUILTIN;
+​
+// Constructs a BSSRDF for subsurface scattering within a homogeneous medium.
+//
+//  \param  N           Normal vector of the surface point beeing shaded.
+//  \param  albedo      Surface albedo.
+//  \param  sss_depth   Mean-free path in units of scene length.
+//  \param  sss_color   Scattering color / transmittance. The desired color resulting from white light transmitted a distance of 'sss_depth'
+//                      through the surface.
+//  \param  anisotropy  Scattering anisotropy [-1,1]. Negative values give backwards scattering, positive values give forward scattering, 
+//                      and 0.0 gives uniform scattering.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color subsurface_bssrdf(normal N, color albedo, float sss_depth, color sss_color, float anisotropy) BUILTIN;
```

Comment:
- Is the supplied albedo really intended to be the "surface" albedo, or the scattering albedo (i.e. scattering coefficient divided by absorption coefficient, of the SSS medium)?   In (ADSK) Arnold the scattering albedo is specified (the standard surface parameter is called "subsurface_color").  The surface exit/entry events are modelled as diffuse white transmission lobes (which in itself is an approximation ignoring the surrounding layer structure, but a reasonable one).

- Are we supplying the raw volumetric parameters of the medium, rather than artist friendly "effective" colors? There is some albedo remapping in (ADSK) Arnold which tries to ensure that the supplied scattering albedo roughly matches the resulting color.

- Nitpick: it should probably be "mean-free-path", not "mean-free path".

Reply:
Great points, thanks @portsmouth. The first point is a copy-paste error in the docs and I'll fix that.

I agree it would be more clear to use the word "transmission", so maybe we should just change both names to transmission_depth and transmission_color. It feels artist friendly and intuitive to say this is the expected color of light transmitted this distance through the medium.

For the third point I personally like it if we can keep it as artist friendly as possible. I have a feeling this discussion about volumetric closures will result in an update of the corresponding nodes in the MaterialX library. And at least on that side we want to keep it as artist friendly as possible. So a remapping needed to ensure this is fine by me.